graphdatascience/procedure_surface/api/graph_sampling_endpoints.py

```diff
@@ -0,0 +1,164 @@
+        relationship_types: Optional[List[str]] = None,
+        node_labels: Optional[List[str]] = None,
+        sudo: Optional[bool] = None,
+        log_progress: Optional[bool] = None,
+        username: Optional[str] = None,
+        concurrency: Optional[Any] = None,
+        job_id: Optional[Any] = None,
+    ) -> GraphSamplingResult:
+        """
+        Computes a set of Random Walks with Restart (RWR) for the given graph and stores the result as a new graph in the catalog.
+
+        This method performs a random walk, beginning from a set of nodes (if provided),
+        where at each step there is a probability to restart back at the original nodes.
+        The result is turned into a new graph induced by the random walks and stored in the catalog.
+
+        Parameters
+        ----------
+        G : Graph
+            The input graph on which the Random Walk with Restart (RWR) will be
+            performed.
+        graph_name : str
+            The name of the new graph in the catalog.
+        start_nodes : list of int, optional
+            A list of node IDs to start the random walk from. If not provided, all
+            nodes are used as potential starting points.
+        restart_probability : float, optional
+            The probability of restarting back to the original node at each step.
+            Should be a value between 0 and 1. If not specified, a default value is used.
+        sampling_ratio : float, optional
+            The ratio of nodes to sample during the computation. This value should
+            be between 0 and 1. If not specified, no sampling is performed.
+        node_label_stratification : bool, optional
+            If True, the algorithm tries to preserve the label distribution of the original graph in the sampled graph.
+        relationship_weight_property : str, optional
+            The name of the property on relationships to use as weights during
+            the random walk. If not specified, the relationships are treated as
+            unweighted.
+        relationship_types : Optional[List[str]], default=None
+            The relationship types used to select relationships for this algorithm run.
+        node_labels : Optional[List[str]], default=None
+            The node labels used to select nodes for this algorithm run.
+        sudo : bool, optional
+             Override memory estimation limits. Use with caution as this can lead to
+            memory issues if the estimation is significantly wrong.
+        log_progress : bool, optional
+            If True, logs the progress of the computation.
+        username : str, optional
+            The username to attribute the procedure run to
+        concurrency : Any, optional
+            The number of concurrent threads used for the algorithm execution.
+        job_id : Any, optional
+            An identifier for the job that can be used for monitoring and cancellation
```

Comment:
and this is copied from the manual as well? I wonder if I should review it or not 🤔 if I have a comment on this, it should be carried over back to the manual I suppose.

Reply:
ok it isn't directly copied